File: src/main/java/tfg/proyecto/TFG/dtos/DTOCarritoItemBajada.java
package tfg.proyecto.TFG.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DTOCarritoItemBajada {
    private Long id;
    private int cantidad;
    private double precioUnitario;
    private Long eventoId;
    private String nombreEvento;
}

File: appMovilShowPass/app/src/main/java/com/example/appmovilshowpass/data/remote/dto/DTOCarritoItemBajada.kt
package com.example.appmovilshowpass.data.remote.dto

import com.example.appmovilshowpass.model.CarritoItem

data class DTOCarritoItemBajada(
    val id: Long,
    val cantidad: Int,
    val precioUnitario: Double,
    val eventoId: Long,
    val nombreEvento: String,
    val imagenEvento: String? = null
)

fun DTOCarritoItemBajada.toCarritoItem(): CarritoItem {
    return CarritoItem(
        id = id,
        cantidad = cantidad,
        precioUnitario = precioUnitario,
        eventoId = eventoId,
        nombreEvento = nombreEvento,
        imagenEvento = imagenEvento

    )
}